zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStoreImpl.java

```diff
@@ -667,11 +667,12 @@ public void restore(final String snapshotId, final Map<String, Path> snapshotFil
     }
   }
 
-  private void copyNamedFileToDirectory(
+  private void moveNamedFileToDirectory(
       final String name, final Path source, final Path targetDirectory) {
     final var targetFilePath = targetDirectory.resolve(name);
     try {
       Files.move(source, targetFilePath);
```

Comment:
:wrench: I would add `ATOMIC_MOVE` as copy option. If that's not supported (a `AtomicMoveNotSupportedException` is raised) then we can fallback to your proposal. 
If the file systems are the same, the move should be atomic?